### PR TITLE
DAOS-11885 bio: various bug fixes

### PR DESCRIPTION
**Pull request by NiuYawei**
1. Skip WAL commit for sysdb temporarily until we figure out a good solution for the sysdb sharing issue.

2. Store SMD pool table for RDB pools separately, otherwise, the RDB pool ID will be conflicting with the regular pool ID.

3. Avoid passing zero meta size to bio_mc_create() when creating pool with existing pool file.

4. In bio_blob_create(), the ABT eventual needs be reset before reusing.

5. In dss_srv_handler(), set dmi_xstream before initializing the sysdb.

Required-githooks: true

Signed-off-by: Niu Yawei <yawei.niu@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/vos/vos_internal.h
Comment (wangshilong) [inline src/vos/vos_internal.h:200]: See vp_excl, could be vp_rdb:1

File: src/vos/vos_pool.c
Comment (wangshilong) [inline src/vos/vos_pool.c:80]: maybe could add path_size as an arg, so we don't need call stat twice.

**Review by gnailzenh — APPROVED** [review]
because this changes a lot of common code, let's wait for CI results before landing.

**Review by jolivier23 — APPROVED** [review]

File: src/vos/vos_internal.h
Comment (NiuYawei) [reply src/vos/vos_internal.h:200]: I'd keep it as separate 'bool', otherwise, the pool open count bits needs be reduced.

File: src/vos/vos_pool.c
Comment (NiuYawei) [reply src/vos/vos_pool.c:80]: right, let's cleanup it in follow-on patches.

**Comment by NiuYawei** [comment]
Only one knowon osa online extending failure, the PR is good for landing.

**Pull request merged**
